src/components/forms/FileInput/FileInput.tsx

```diff
@@ -12,164 +12,186 @@ interface FileInputProps {
   accept?: string
   onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
   onDrop?: (e: React.DragEvent) => void
-  inputRef?: React.RefObject<HTMLInputElement>
 }
 
-export const FileInput = ({
-  name,
-  id,
-  disabled,
-  multiple,
-  className,
-  accept,
-  onChange,
-  onDrop,
-  inputRef,
-  ...inputProps
-}: FileInputProps & JSX.IntrinsicElements['input']): React.ReactElement => {
-  const [isDragging, setIsDragging] = useState(false)
-  const [showError, setShowError] = useState(false)
-  const [files, setFiles] = useState<FileList | null>(null)
-
-  const fileInputClasses = classnames(
-    'usa-file-input',
-    {
-      'usa-file-input--disabled': disabled,
-    },
-    className
-  )
-
-  const targetClasses = classnames('usa-file-input__target', {
-    'usa-file-input--drag': isDragging,
-    'has-invalid-file': showError,
-  })
-
-  const hideDragText =
-    /rv:11.0/i.test(navigator.userAgent) ||
-    /Edge\/\d./i.test(navigator.userAgent)
-
-  const dragText = multiple ? 'Drag files here or ' : 'Drag file here or '
-
-  const filePreviews = []
-  if (files) {
-    for (let i = 0; i < files?.length; i++) {
-      const imageId = makeSafeForID(files[parseInt(`${i}`)].name)
-      const key = `filePreview_${imageId}`
-      filePreviews.push(
-        <FilePreview
-          key={key}
-          imageId={imageId}
-          file={files[parseInt(`${i}`)]}
-        />
-      )
-    }
-  }
+export type FileInputRef = {
+  clearFiles: () => void
+  input: HTMLInputElement | null
+  files: FileList | []
+}
```

Comment:
A couple things:
- Make sure we export this type from `index.ts`. Its not right now.
- `files` types `FileList` or `[]` was causing me issues (type errors, not wanting to handle the empty array).  In my understanding, a `FileList` can be empty. Why do we have an empty array type here as well?

Reply:
Ok, I'm pushing a change that modifies `FileInput` so that `files` state is an array of files, rather than a `FileList` interface. I think that will resolve the issue of how to handle an "empty" file state, given you cannot construct or modify a `FileList`.

fun fact for reference, the w3c spec says the FileList interface is "at-risk" of being replaced with an array anyways! https://w3c.github.io/FileAPI/#filelist-section

Reply:
Ohh okay that's good to know.